src/main/java/com.company/tree/SumOfLongRootToLeafPath.java

```diff
@@ -0,0 +1,63 @@
+package com.company.tree;
+
+public class SumOfLongRootToLeafPath {
+    int maxSum = Integer.MIN_VALUE;
+    int maxPathLength = Integer.MIN_VALUE;
+    public int sumOfLongRootToLeafPath(TreeNode root)
+    {
+        sumOfLongRootToLeafPathRecursive(root, 0, 0);
+        return maxSum;
+    }
+
+    private void sumOfLongRootToLeafPathRecursive(TreeNode node, int sum, int pathLength) {
+        if(node == null) {
+            return;
+        }
+        sum = sum + node.val;
+        pathLength = pathLength + 1;
+        sumOfLongRootToLeafPathRecursive(node.left, sum, pathLength);
+        sumOfLongRootToLeafPathRecursive(node.right, sum, pathLength);
+        if(pathLength > maxPathLength) {
```

Comment:
This can be written as single if condition. 
if (pathLength >= maxPathLength) {
    maxPathLength = pathLength;
    maxSum = Math.max(maxSum, sum);
}

Reply:
okay